### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by maxux**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+	jwtPubKey, err = jwt.ParseECPublicKeyFromPEM([]byte(iyoPubKey))
+	if err != nil {
+		log.Panicf("failed to parse pub key:%v\n", err)
+	}
+}
+
+//
+// list of sessions
+//
+func NewSessions(backend *ZdbBackend) *Sessions {
```
Comment (zaibon) [inline main.go:60]: In Go, comment on function of method always starts with the name of the func/method you document. 
So here that should be for example
```go
//NewSessions creates a Sessions object used to keep track of all active sessions of the server
func NewSessions(backend *ZdbBackend) *Sessions {
```

This apply for all function documentation

File: main.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+	return &Sessions{
+		List:    make(map[*resp.Conn]*Session),
+		Backend: backend,
+	}
+}
+
+// single session
+func (ss *Sessions) NewSession() *Session {
+	return &Session{
+		Authenticated: false,
```
Comment (zaibon) [inline main.go:72]: False and empty string are the default value for boolean and string, so you don't have to write them explicitly. Not that explicit it bad, just FYI

File: main.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		return jwtPubKey, nil
+	})
+
+	if err != nil {
+		return false, "", err
+	}
+
+	if claims, ok := token.Claims.(jwt.MapClaims); ok && token.Valid {
```
Comment (zaibon) [inline main.go:100]: How exactly are you verifying anything here ?
I don't see you comparing value in the claims to anything.

File: main.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+	conn.WriteString("Gateway for 0-db (zdb) compatible clients\n")
+	return true
+}
+
+//
+// EXISTS
+//
+func (ss *Sessions) Exists(conn *resp.Conn, args []resp.Value) bool {
+	// retreive session from connection
+	session := ss.List[conn]
```
Comment (zaibon) [inline main.go]: you should probably handle the case where the connection doesn't exist in the map, otherwise this will panic

File: main.go [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+
+	return true
+}
+
+//
+// SET
+//
+func (ss *Sessions) Set(conn *resp.Conn, args []resp.Value) bool {
+	// retreive session from connection
+	session := ss.List[conn]
```
Comment (zaibon) [inline main.go]: same remark you should probably handle the case where the connection is not in the map

File: main.go [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+	return session.Set(conn, args)
+}
+
+func (s *Session) DataValidator(key []byte, payload []byte) bool {
```
Comment (zaibon) [inline main.go]: This can be a function since your don't use anything from the Session object

**Review by zaibon — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (maxux) [reply main.go:72]: Thanks for the notice

File: main.go (hunk 3 above)
Comment (maxux) [reply main.go:100]: According to the documentation, this is how to validate a jwt.
https://godoc.org/github.com/dgrijalva/jwt-go#example-Parse--Hmac

**Comment by maxux** [comment]
Thanks for the comments @zaibon !

**Pull request merged**